src/HypothesisClient/ApiClient/ApiClientTrait.php

```diff
@@ -36,24 +52,36 @@ public function __construct(HttpClient $httpClient, array $headers = [])
     final protected function postRequest(
         UriInterface $uri,
         array $headers,
-        StreamInterface $content
+        string $content
     ) : PromiseInterface {
         $headers = array_merge($this->headers, $headers);
 
-        $request = new Request('DELETE', $uri, $headers, $content);
+        $request = new Request('POST', $uri, $headers, $content);
 
         return $this->httpClient->send($request);
     }
 
     final protected function putRequest(
         UriInterface $uri,
         array $headers,
-        StreamInterface $content
+        string $content
     ) : PromiseInterface {
         $headers = array_merge($this->headers, $headers);
 
         $request = new Request('PUT', $uri, $headers, $content);
 
         return $this->httpClient->send($request);
     }
+
+    final protected function patchRequest(
+        UriInterface $uri,
+        array $headers,
+        string $content
+    ) : PromiseInterface {
+        $headers = array_merge($this->headers, $headers);
```

Comment:
Maybe have a `createRequest()` method to avoid duplicating this?

Reply:
👍